### PR TITLE
Testing works again

### PR DESCRIPTION
**Pull request by phg49389**
There’s still one error to fix: searchFiles

But Caleb said that someone else can get that fixed. The other 98% of the testing suite works, including the Travis CI stuff.

### CODE REVIEW

**Comment by phg49389** [comment]
This PR should resolve Issues #304 and #307. Once it gets merged in, then we can start work on #342

**Comment by c-schwarze** [comment]
@j-jasik when you have a chance after the tutorlabs launch, lets take a look at this.

After we get this up and running for this PR, I want you to help Bailey create tests to get this working with master.

File: testing_programs.csv [hunk 1]
```diff
@@ -0,0 +1,12900 @@
+key,tag,outcome,other,topic
```
Comment (c-schwarze) [inline testing_programs.csv]: We definitely don't want this in version control.

Comment (phg49389) [reply testing_programs.csv]: Ok 👍 

I'll take a look at how to excise it from Git while still letting it work in Travis CI

**Comment by phg49389** [comment]
I think `testing_app.db` has been part of the testing suite for a while. Was it removed by something? Did we talk about removing it?

File: testing_programs.csv (hunk 1 above)
Comment (c-schwarze) [reply testing_programs.csv]: thats not what i'm saying. I mentioned I was going to work on this. Also, I was saying we should limit this file, as we don't need this large of a test file. Also, it has "real" data in it that we don't want shown to the world

**Comment by c-schwarze** [comment]
We haven't talked about removing it. But I think we need it reduced, since its huge and it has real data

**Pull request merged**